src/caip-types.test-d.ts

```diff
@@ -0,0 +1,52 @@
+import { expectAssignable, expectNotAssignable } from 'tsd';
+
+import type {
+  CaipAccountAddress,
+  CaipAccountId,
+  CaipChainId,
+  CaipNamespace,
+  CaipReference,
+} from '.';
+
+const embeddedString = 'test';
+
+// Valid caip strings:
+
+expectAssignable<CaipChainId>('namespace:reference');
+expectAssignable<CaipChainId>('namespace:');
+expectAssignable<CaipChainId>(':reference');
+expectAssignable<CaipChainId>(`${embeddedString}:${embeddedString}`);
+
+expectAssignable<CaipNamespace>('string');
+expectAssignable<CaipNamespace>(`${embeddedString}`);
+
+expectAssignable<CaipReference>('string');
+expectAssignable<CaipReference>(`${embeddedString}`);
+
+expectAssignable<CaipAccountId>('namespace:reference:accountAddress');
+expectAssignable<CaipAccountId>('namespace:reference:');
```

Comment:
Similar as above — I'm assuming this is non-enforceable via TypeScript?

Reply:
See above